ext/odbc/php_odbc.c

```diff
@@ -2652,36 +2677,42 @@ PHP_FUNCTION(odbc_errormsg)
 */
 PHP_FUNCTION(odbc_setoption)
 {
-	odbc_connection *conn;
+	odbc_link *link;
 	odbc_result	*result;
 	RETCODE rc;
 	zval *pv_handle;
 	zend_long pv_which, pv_opt, pv_val;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "rlll", &pv_handle, &pv_which, &pv_opt, &pv_val) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "olll", &pv_handle, &pv_which, &pv_opt, &pv_val) == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	switch (pv_which) {
 		case 1:		/* SQLSetConnectOption */
```

Comment:
Are these constants that we define? If not it may make sense to expose them so the error message is less weird?

Or even better, split this function into two dedicated ones.

Reply:
yeah, honestly, this could have been tackled by my overloaded function RFC... but the possibility has gone so I'll try to address this comment in a followup

Reply:
Yup this can be done in a follow-up :D